Core/include/Acts/TrackFitting/detail/GsfUtils.hpp

```diff
@@ -91,22 +92,29 @@ class ScopedGsfInfoPrinterAndChecker {
 
   void checks(const std::string_view &where) const {
     const auto cmps = m_stepper.constComponentIterable(m_state.stepping);
-    throw_assert(detail::weightsAreNormalized(
-                     cmps, [](const auto &cmp) { return cmp.weight(); }),
-                 "not normalized at " << where);
-
-    throw_assert(
-        std::all_of(cmps.begin(), cmps.end(),
-                    [](auto cmp) { return std::isfinite(cmp.weight()); }),
-        "some weights are not finite at " << where);
+
+    // If all are missed, weights have been reset to zero, so it might not be
+    // normalized
```

Comment:
Maybe you could be more explicit in this comment:
- `If all are missed` - `all` what?
- `it might not be normalized` - what is `it`?

Reply:
Yeah the comment is not actually expressive I admit...
all = all components
it = the sum of all components
but I should improve this comment, you're right!

Reply:
@AJPfleger I improved the comment, if it makes sense to you you could approve